cornice_swagger/swagger.py

```diff
@@ -383,6 +393,86 @@ def _ref(self, param, depth, base_name=None):
         return {'$ref': pointer}
 
 
+class ResponseHandler(object):
+    """Handles swagger response definitions."""
+
+    json_pointer = '#/responses/'
+
+    def __init__(self, definition_handler=DefinitionHandler(), ref=False):
+        """
+        :param definition_handler:
+            Callable that handles swagger definition schemas.
+        :param ref:
+            Specifies the ref value when calling from_xxx methods.
+        """
+
+        self.responses = {}
+
+        self.definitions = definition_handler
+        self.ref = ref
+
+    def from_schema_mapping(self, schema_mapping):
+        """
+        Creates a Swagger response object from a dict of response schemas.
+
+        :param schema_mapping:
+            Dict with entries matching ``{status_code: response_schema}``.
+        :rtype: dict
+            Response schema.
+        """
+
+        responses = {}
+
+        for status, response_schema in schema_mapping.items():
+
+            response = {}
+            if response_schema.description:
+                response['description'] = response_schema.description
+            else:
+                raise CorniceSwaggerException('Responses must have a description.')
+
+            for field_schema in response_schema.children:
+                location = field_schema.name
+                if location == 'body':
+                    field_schema.title = field_schema.__class__.__name__
+                    response['schema'] = self.definitions.from_schema(field_schema)
+                elif location == 'headers':
+                    headers = convert_schema(field_schema)
+                    if 'properties' in headers:
+                        response['headers'] = headers['properties']
+
+            pointer = response_schema.__class__.__name__
+            response = self._ref(response, self.ref, pointer)
+            responses[status] = response
+
+        return responses
+
+    def _ref(self, resp, depth, base_name=None):
+        """
+        Map responses in the responses section using JSON pointers.
+
+        :param schema:
+            Swagger response definition.
+        :param depth:
+            If param should be set as a reference. True means it should.
+        :param base_name:
+            Name that should be used for the reference.
+
+        :rtype: dict
+            Swagger schema.
```

Comment:
- I found the description of this method confusing, maybe because `Map` is a pretty generic verb. (Too much Haskell?) If I understood this method correctly, I would describe it as "Store a response schema definition and return a reference to it", maybe with the word "conditionally" in there somewhere.
- `depth` corresponds to `resp_ref` in the `CorniceSwagger`, so it almost never changes, right? A name like `depth` made me think that for some responses, the depth would be deeper, and then it would use them more, or something. Should we even take this as an argument vs. making the call to this method conditional (since I think it only occurs once)?
- "Swagger schema" for `dict` is pretty vague. `A JSON pointer to the original response definition, or the original definition if depth was True` would be better.